sqlserver/assets/configuration/spec.yaml

```diff
@@ -39,10 +39,58 @@ files:
       value:
         type: string
     - name: database
-      description: Database name to query
+      description: |
+        Database name to query.  Not compatible with `database_autodiscovery`.
       value:
         type: string
         example: master
+    - name: database_autodiscovery
+      description: |
+        Auto-discover and monitor databases.  If `true`, overrides `database` option.
+        Can be combined with `autodiscovery_include` and `autodiscovery_exclude` options.
+      value:
+        type: boolean
+        example: false
+    - name: autodiscovery_include
+      description: |
+        Regular expression for specific database names to always include as part of `database_autodiscovery`.
+        Will only report metrics for databases that are actually found in this instance, so missing
+        databases from this option should be harmless.
+
+        Character casing is ignored. For convenience, the regular expressions start matching from the beginning and
+        therefore to match anywhere you must prepend `.*`. For exact matches append `$`. Casing is ignored.
+
+        When conflicts arise, included databases will take precedence.
```

Comment:
I think it should be the other way around, and have `excluded` take precedence as in other integrations, see disk https://github.com/DataDog/integrations-core/blob/7296b06af880e80277ad96956d858f89b4d46c1b/disk/datadog_checks/disk/data/conf.yaml.default#L102 and vsphere https://github.com/DataDog/integrations-core/blob/7296b06af880e80277ad96956d858f89b4d46c1b/vsphere/datadog_checks/vsphere/data/conf.yaml.example#L114-L115

I don't see a case where having `include` > `exclude` would make the configuration easier than `exclude` > `include`, but maybe I'm missing something here.

Reply:
Great point, my thinking was the current way would have benefits for a database admin, but it would make more sense to be consistent with our other approaches.

Reply:
Approach updated, along with corresponding tests.